articles/stripe-integration-react/index.md

```diff
@@ -0,0 +1,139 @@
+![Dashboard View](/engineering-education/stripe-integration-react/dashboard.jpg)
+*Dashboard View*
+
+#### Obtain a Stripe Access Token
+You will find a section called `Get your test API keys`. Under there, you will find two keys, a **Publishable key**, and a **Secret key**. As the name suggests, we need to keep these keys secure. The publishable key is used to send requests, whereas the secret key is stored in the `.env` file at the backend.
+
+![API Test Keys](/engineering-education/stripe-integration-react/api_test_keys.jpg)
+*API Keys on Dashboard*
+
+### Stripe Integration using react-stripe-checkout
+There are many libraries available for integrating React with stripe. We will be using the `react-stripe-checkout` library in this tutorial.
+
+Install the library using the following command:
+
+```
+npm install react-stripe-checkout
+```
+
+Once installed, create a new component called `CheckoutWithStripe.js` and add it to your main component list. Insert the appropriate details and create your custom functions to add more details to the API call.
+
+Here is an example function that you can modify using your custom store details:
+
+```jsx
+import React from 'react';
+import StripeCheckout from 'react-stripe-checkout';
+
+// sample function defined to compute total quantity of cart
+function computeQuantity(cart) {
+    return cart.reduce((count, itemInCart) => count + itemInCart.quantity, 0);
+}
+// similar functions can be defined to compute total price, email of the user, etc.
+
+class CheckoutWithStripe extends React.Component {
+    onToken = (res) => {
+        fetch('/save-stripe-token', {
+            method: 'POST',
+            body: JSON.stringify(token),
+        }).then(res => {
+            res.json().then(data => {
+                console.log(`Payment token generated, ${data.name}`)
+            })
+        })
+    };
+
+    render() {
+        return (
+            <StripeCheckout
+                amount = '10.00'
+                name="STRIPE_INTEGRATION"
+                // functions defined above can be used to add more information while making the API call.
+                // description={`Order of ${computeQuantity(cart)} items!`}
+                image='LINKTOIMAGE'
+                stripeKey="PUBLISHABLE_STRIPE_KEY"
+                currency="INR"
+                email='USER_EMAIL'
+                token={this.onToken}/>          
+        );
+    }
+}
+
+export default CheckoutWithStripe
+```
+
+Call the component `CheckoutWithStripe` from the payment page.
+
+You should see a button which says `Pay With Card`.
+
+An example is given below:
+![Paywithcard stripe](/engineering-education/stripe-integration-react/paywithcard.jpg)
+*Intermediate Output*
+
+Finally, a dialog box should appear to enter card details:
+
+![Final Output](/engineering-education/stripe-integration-react/final_output.jpg)
+*Final Output*
+
+The error says that the API key is invalid.
+
+To resolve the error, you may enter your public API key.
```

Comment:
I would reconstruct your tutorial so that the reader inputs the public API key earlier so that they don't get this error. Otherwise, this tutorial comes off as incomplete...

Also, I think this tutorial should be rewritten so that this sentence is not present, but if it has to stay in then you must expound on it. Where/how do they "enter your public API key"?

Reply:
The reason I have not included my publishable key is for security reasons! I have mentioned PUBLISHABLE_KEY in the code and how to replace it with the code obtained from dashboard.